lib/players/ai.rb

```diff
@@ -7,11 +7,18 @@ def initialize(checker, token, min_token)
     @min_token = min_token
   end
 
-  def get_move(grid, turns_remaining)
+  def get_move(grid, turns_remaining, user_interface)
+    user_interface.display_message("\nThe computer is picking a spot...")
+    user_interface.pause
     move = minimax(grid, turns_remaining, -Float::INFINITY, Float::INFINITY, true)[1]
     return move
   end
 
+  def announce_move(move, user_interface)
+    user_interface.display_message("\nThe computer picks spot #{move + 1}.")
+    user_interface.pause
+  end
+
```

Comment:
Since the user interface is used in two separate methods, and coupled with the fact that `announce_move` is a pure function, I think its worth exploring a different way to handle this scenario. Off the bat it feels like the user interface should be an instance property on the AI, but now I'm thinking about the idea of "anouncing" a move, and wondering if it really fits in the domain of a move picking strategy, or if it should bubble up to being a responsibility on the Player type.

Reply:
When I started doing it, I liked the idea of it being on the player type but was trying to deal with accommodating different prompts based on what the behavior was.  I think I have an idea to make it work on the player.  I'll give it a go and see.

Reply:
OK, I made some adjustments.  I added a behavior script to the behaviors, which I can reference from player.  Debated where to include it: on behavior or passed into player when in app.  Figured that might complicate the upcoming ability to pick modes, etc.  Let me know what you think.